src/classlibnative/float/floatdouble.cpp

```diff
@@ -0,0 +1,299 @@
+FCIMPL1_V(double, COMDouble::Cosh, double d) 
+    FCALL_CONTRACT;
+
+    return (double)cosh(d);
+FCIMPLEND
+
+/*=====================================Tanh====================================
+**
+==============================================================================*/
+FCIMPL1_V(double, COMDouble::Tanh, double d) 
+    FCALL_CONTRACT;
+
+    return (double)tanh(d);
+FCIMPLEND
+
+/*=====================================ModFDouble==============================
+**
+==============================================================================*/
+FCIMPL1(double, COMDouble::ModFDouble, double* pDblValue)
+    FCALL_CONTRACT;
+
+    return (double)modf(*pDblValue, pDblValue);
+FCIMPLEND
+
+// Log, Log10 and Exp were previously slower with /fp:fast on SSE2 enabled hardware (see #500373).
+// This is no longer the case and they now consume use the /fp:fast versions.
+
+/*=====================================Log======================================
+**This is the natural log
+==============================================================================*/
+FCIMPL1_V(double, COMDouble::Log, double d) 
+    FCALL_CONTRACT;
+
+    return (double)log(d);
+FCIMPLEND
+
+/*====================================Log10=====================================
+**This is log-10
+==============================================================================*/
+FCIMPL1_V(double, COMDouble::Log10, double d) 
+    FCALL_CONTRACT;
+
+    return (double)log10(d);
+FCIMPLEND
+
+// Exp(+/-INFINITY) did not previously return the expected results of +0.0 (for -INFINITY)
+// and +INFINITY (for +INFINITY) so these cases were handled specially. As this is no longer
+// the case and the expected results are now returned, the special handling has been removed.
+
+/*=====================================Exp======================================
+**
+==============================================================================*/
+FCIMPL1_V(double, COMDouble::Exp, double d) 
+    FCALL_CONTRACT;
+
+    return (double)exp(d);
+FCIMPLEND
+
+// Previously there was more special handling for the x86 Windows version of Pow.
+// This additional handling was unnecessary and has since been removed.
+
+/*=====================================Pow======================================
+**
+==============================================================================*/
+FCIMPL2_VV(double, COMDouble::Pow, double x, double y)
+{
+    FCALL_CONTRACT;
+
+    double result;
+
+    // The CRT version of pow preserves the NaN payload of x over the NaN payload of y.
+
+    if(_isnan(y)) {
+        return y; // IEEE 754-2008: NaN payload must be preserved
+    }
+
+    if(_isnan(x)) {
+        return x; // IEEE 754-2008: NaN payload must be preserved
+    }
+
+    // The CRT version of pow does not return NaN for pow(-1.0, +/-INFINITY) and
+    // instead returns +1.0.
+
+    if(IS_DBL_INFINITY(y) && IS_DBL_NEGATIVEONE(x)) {
+        *((INT64 *)(&result)) = CLR_NAN_64;
```

Comment:
Just a question, not too familiar with native code but isn't violating strict aliasing [supposed to be UB](http://stackoverflow.com/questions/98650/what-is-the-strict-aliasing-rule)?